File: Leetcode/bestTimetoBuyandSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minProfit = 99999;
        int size = prices.length;
        for (int i = 0; i < size; i++) {
            if(prices[i] < minProfit) {
                minProfit = prices[i];
            }
            
            int tempProfit = prices[i] - minProfit;
            if (tempProfit > maxProfit) {
                maxProfit = tempProfit;
            }
        }
        return maxProfit;
    }
}

File: neetcode/sliding_window/best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int size = prices.size();
        int maxIdx = 0;
        int maxProfit = 0;
        int left = 0, right = 0;
        while(left < size && right < size){
            if(prices[right] < prices[left]) {
                left = right;
                maxIdx = left;
            }
            if(prices[right] >= prices[maxIdx]) maxIdx = right;
            right++;
            int curProfit = prices[maxIdx] - prices[left];
            maxProfit = max(maxProfit, curProfit);
        }

        return maxProfit;
    }
};
